R/methods-Megastudy.R

```diff
@@ -271,11 +271,7 @@ setMethod('getDTWithImputedZeroes', signature = c('Megastudy', 'VariableMetadata
   upstreamEntityVariables.dt <- unique(upstreamEntityVariables.dt)
   veupathUtils::logWithTime(paste("Found", nrow(upstreamEntityVariables.dt), "unique existing upstream variable value combinations."), verbose)
   if (!!length(collectionsDT)) {
-    upstreamEntityVariables.dt <- merge(upstreamEntityVariables.dt, collectionsDT, by=studyEntityIdColName, all=TRUE, allow.cartesian=TRUE)
-  upstreamEntityVariables.dt <- upstreamEntityVariables.dt[, -which(grepl('.x',names(upstreamEntityVariables.dt),fixed=T)), with=FALSE]
-    collectionEntityColumnIndices <- which(grepl('.y',names(upstreamEntityVariables.dt),fixed=T))
-    names(upstreamEntityVariables.dt)[collectionEntityColumnIndices] <- gsub('.y', '', names(upstreamEntityVariables.dt)[collectionEntityColumnIndices], fixed=T)
-    upstreamEntityVariables.dt <- unique(upstreamEntityVariables.dt)
+    upstreamEntityVariables.dt <- collectionsDT
```

Comment:
I was just wondering why no more fancy merge?

Naively I wonder if anything upstream of collection (study, collection site) might be discarded?

Reply:
the 'fancy' merge was basically me tossing out variables on the sample data table anyway..